industrydive/wavepool
# News Post detail page bug fixes

The following issues have been resolved:
* gitignore - populated with general python related values.  Also now excludes VSCode related settings.
* Edit button - now properly hidden from non-authenticated users & fixed the link's ID so the test actually tests something.
* Edit button - now links to the edit screen for the post in the admin section for authenticated users
* Page Title - now includes the newspost title.
* Divesite Display Name - Fixed the `source_divesite_name` property to parse the site url to grab the TLD and return the associated Divesite Name.  Currently assumes that the url will be along the lines of "http://www.[divesite].com/rest-of-the-url"

The `test_newspost_page_content` test had a lot of issues that needed to be cleaned up, most of them are now.  The majority of the fixtures pass through just fine.  The one area that needs to be decided how to handle is when the body includes broken HTML (in this case, we have a `</p>` without a matching `<p>`).  Most browsers will still display this fairly ok, but it may not display how the user expected it to.  My preferred way to handle this would be adding validation when the field is populated, but this may not be feasible if the data is gathered via automation.  In that case, we should probably allow it in, but notify someone that the record includes invalid HTML.  A third possibility is to just not care (obviously this would be easiest, but not recommended).

### Testing
* Run `python manage.py test wavepool.tests.NewsPostDetail`.  All tests should pass except for `test_newspost_page_content`.  This will fail due to the fixture for post id 3 for the reason noted above.
* Browse to any post's page without logging in, you should no longer see the edit button.
* Each page should now also display the correct dive site name after "See the live story at "
* Verify page title in the browser tab now includes the newspost title.  Verify that frontpage does not include this.
* Login to Admin, then browse to one of the detail pages.  Verify that the Edit button shows and that clicking it takes you to the correct edit screen in the Admin.
